lightning/src/ln/channelmanager.rs

```diff
@@ -13662,14 +13634,20 @@ where
 						}
 					}
 				}
+				let mut per_peer_state =  per_peer_state.get(counterparty_node_id)
+					.expect("If we have pending updates for a channel it must have an entry")
+					.lock().unwrap();
 				if updated_id {
-					per_peer_state.get(counterparty_node_id)
-						.expect("If we have pending updates for a channel it must have an entry")
-						.lock().unwrap()
+					per_peer_state
 						.closed_channel_monitor_update_ids.entry(*channel_id)
 						.and_modify(|v| *v = cmp::max(update.update_id, *v))
 						.or_insert(update.update_id);
 				}
+				let in_flight_updates = per_peer_state.in_flight_monitor_updates
+					.entry(*funding_txo)
+					.or_insert_with(Vec::new);
+				debug_assert!(!in_flight_updates.iter().any(|upd| upd == update));
+				in_flight_updates.push(update.clone());
```

Comment:
Missing test coverage for this

Reply:
Oh, I had commented the lines out. Probably all tests passed because updates are duplicated in pending background events?